### PR TITLE
[Rooms] Update quickstart to use new sdk version

### PR DESCRIPTION
**Pull request by Mrayyan**
## Purpose

Updated Rooms quickstart project to use latest version.

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by minnieliu — APPROVED** [review]

File: rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java [hunk 1]
```diff
@@ -72,128 +70,118 @@ public static boolean deleteRoom(String roomId)
...
                 System.out.println("1. Add a room");
                 System.out.println("2. Update a room");
                 System.out.println("3. Delete a room");
                 System.out.println("4. Get room details");
-                System.out.println("5. List room ids");
-                System.out.println("6. Add a participant");
-                System.out.println("7. Update a participant");
+                System.out.println("5. List room ids created in this session");
+                System.out.println("6. Add or Update participants");
+                System.out.println("7. List rooms in resource");
```
Comment (mikehang-msft) [inline rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java:184]: Can we swap 6 and 7? so we can have Rooms and Participants management separated in two groups.

File: rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java [hunk 2]
```diff
@@ -204,64 +192,54 @@ public static void main(String[] args)
...
 
                         break;
                     }
-                    case 7:
-                    {
-                        System.out.print("RoomId:");
-                        String roomId = br.readLine();
-                        updateParticipant(roomId);
+                    case 7: {
+                        System.out.print("Listing all rooms");
```
Comment (mikehang-msft) [inline rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java:230]: For consistency, either leave this message inside the function itself or here but ensure doing the same for all other functions.

File: rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java (hunk 1 above)
Comment (Mrayyan) [reply rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java:184]: Swaped

File: rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java (hunk 2 above)
Comment (Mrayyan) [reply rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java:230]: Updated

File: rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java [hunk 3]
```diff
@@ -72,128 +70,118 @@ public static boolean deleteRoom(String roomId)
...
-    public static void addParticipant(String roomId)
-    {
+    public static void listRooms() {
         try {
-            RoomParticipant newParticipant = new RoomParticipant()
-                .setCommunicationIdentifier(new CommunicationUserIdentifier(USER_ID_3))
-                .setRole(RoleType.CONSUMER);
-            ParticipantsCollection updatedParticipants = roomsClient.addParticipants(roomId, List.of(newParticipant));
-            System.out.println("Participants: " + listParticipantsAsString(updatedParticipants.getParticipants()));
+            PagedIterable<CommunicationRoom> rooms = roomsClient.listRooms();
```
[the remark this answers is not in the document]
Comment (Mrayyan) [reply rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java:109]: Done, but I did it for the first 3 rooms. Also added a print saying that only first 3 rooms are printing for demo.

File: rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java [hunk 4]
```diff
@@ -7,21 +8,23 @@
...
 import com.azure.communication.common.CommunicationUserIdentifier;
 import com.azure.communication.rooms.RoomsClient;
 import com.azure.communication.rooms.RoomsClientBuilder;
+import com.azure.core.http.rest.PagedIterable;
 import com.azure.core.util.Context;
 
-public class App
-{
+public class App {
     static String USER_ID_1 = "<communication-user-id-1>";
```
[the remark this answers is not in the document]
Comment (Mrayyan) [reply rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java]: Updated now

**Review by mikehang-msft — APPROVED** [review]

File: rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java [hunk 5]
```diff
@@ -72,204 +88,208 @@ public static boolean deleteRoom(String roomId)
...
-            System.out.println("Participants: " + listParticipantsAsString(updatedParticipants.getParticipants()));
+            PagedIterable<CommunicationRoom> rooms = roomsClient.listRooms();
+
+            int count = 0;
+            System.out.println("For this demo only 3 rooms will be printed.");
+
+            for (CommunicationRoom room : rooms) {
+                System.out.println("\nRoom ID: " + room.getRoomId());
+
+                if (count >= 3) {
```
[the remark this answers is not in the document]
Comment (Mrayyan) [reply rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java]: Got it, updated now.

**Review by minnieliu — APPROVED** [review]

**Pull request merged**